app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -423,8 +424,15 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             // that we should not rely on the application being killed between user sessions.
             components.appStore.dispatch(AppAction.ResumedMetricsAction)
 
-            DefaultBrowserNotificationWorker.setDefaultBrowserNotificationIfNeeded(applicationContext)
-            ReEngagementNotificationWorker.setReEngagementNotificationIfNeeded(applicationContext)
+            with(applicationContext) {
+                // Only set up Workers if notifications are enabled
+                val notificationManagerCompat = NotificationManagerCompat.from(this)
+                if (notificationManagerCompat.isNotificationChannelEnabled(MARKETING_CHANNEL_ID)) {
```

Comment:
Triple checking here: has onboarding already run by now? Has the user been asked for permission?

Also: should we be using `areNotificationsEnabledSafe`?

Reply:
Yep 👍 line 336 showNotificationPermissionPromptIfRequired() will ensure the dialog is displayed if needed. 

On a fresh install:
Application is loaded, onResume is called before showNotificationPermissionPromptIfRequired dialog is displayed - will fail at isNotificationChannelEnabled check. This will be followed by the permission dialog which once closed/dismissed will trigger onResume again where we will check for persmission.

areNotificationsEnabledSafe check is used as part of isNotificationChannelEnabled 🙂